src/main/java/in/projecteka/consentmanager/consent/ConsentManager.java

```diff
@@ -111,11 +113,28 @@ private String getHIUId(RequestedDetail requestedDetail) {
         return consentRequestRepository.requestsForPatient(patientId, limit, offset);
     }
 
+    private Mono<Void> validateLinkedHips(String authorizationToken, List<GrantedConsent> grantedConsents) {
+        return patientServiceClient.retrievePatientLinks(authorizationToken)
+                .flatMap(linkedCareContexts ->
+                        Flux.fromIterable(grantedConsents)
+                                .filter(grantedConsent ->
+                                        linkedCareContexts.hasHipReference(grantedConsent.getHip().getId())
+                                                && linkedCareContexts.hasCCReferences(grantedConsent.getHip().getId(),
```

Comment:
> linkedCareContexts.hasCCReferences
This method does both HIP validation and care context validation right, then do we need this check `linkedCareContexts.hasHipReference(grantedConsent.getHip().getId())`

Reply:
no. I will make that method private. it is still used internally.

Reply:
done